### DISCUSSION TITLE
Question about coupling between OpenMC and Tensor Mechanics Module

### DISCUSSION THREAD

**Discussion by WayneX-Wong** [neams-th-coe/cardinal#510] [Q&A] [answered]
Hello everyone,

I'm wondering whether OpenMC can be coupled with Tensor Mechanics Module using Thermal Conduct to calculate temperature ?

Because the strain calculated by Tensor Mechanics means the change of mesh, while as I know, OpenMC in cardinal can't handle this kind of on-line geometry modification so far.

Is there anyway we can couple with Tensor Mechanics Module? Even if there's only one-way coupling of temperature from Thermal Conduct to Tensor Mechanics, while still two-way coupling between OpenMC and Thermal Conduct at the same time?

Thanks!

**Comment by aprilnovak** [comment, marked as answer]
Hi @WayneX-Wong - you are correct, Cardinal currently only supports cell-based OpenMC geometries. We plan to add unstructured mesh geometries this year, which will allow OpenMC to solve on changing geometries.

Yes, you can certainly achieve a mixed two-way/one-way coupling with the existing setup:

- OpenMC and heat conduction are two-way coupled via power and temperatures
- heat conduction and tensor mechanics are two-way coupled via temperatures and geometry
- geometry changes to OpenMC are ignored

To do the above, simply omit any transfer of displacement into the OpenMC-wrapped input file. To learn how to accomplish the OpenMC-heat conduction coupling, you can check out [these tutorials](https://cardinal.cels.anl.gov/tutorials/openmc_solid.html). The [MOOSE website](https://mooseframework.inl.gov) should then have all the other documentation you need to add tensor mechanics to this solve.

**Reply by WayneX-Wong** [reply]
That's very helpful, many thanks for your answer! @aprilnovak

**Reply by mandee123mandee** [reply]
Does Cardinal support unstructured mesh geometries now?

**Reply by aprilnovak** [reply]
We currently supported unstructured surface meshes via DAGMC -- a tutorial can be found here: https://cardinal.cels.anl.gov/tutorials/dagmc.html

We have ongoing work to add volumetric unstructured meshes, and I can post an update here once it is ready.
